src/codegen/dead_storage.rs

```diff
@@ -92,8 +102,20 @@ fn reaching_definitions(cfg: &mut ControlFlowGraph) -> (Vec<Vec<Vec<Transfer>>>,
         let mut vars = if let Some(vars) = block_vars.get(&block_no) {
             vars[0].clone()
         } else {
+            assert_eq!(block_no, 0);
+
+            let mut vars: HashMap<usize, HashMap<InstrDef, Option<Expression>>> = HashMap::new();
+
+            for var_no in cfg.vars.keys() {
+                let mut defs = HashMap::new();
+
+                defs.insert(InstrDef::Undefined, None);
+
+                vars.insert(*var_no, defs);
```

Comment:
Why don't `vars.insert(*var_no, HashMap::new())`?

Reply:
That won't work because we want a HashMap with a single `Undefined` entry. I don't think there is a macro for hashmap like there is for `Vec`: `vec![1]`.